libs/Microsoft.MixedReality.WebRTC/PeerConnection.cs

```diff
@@ -309,7 +309,23 @@ internal class DelayedEvent
         /// <summary>
         /// The event handler.
         /// </summary>
-        public Action Event;
+        public Action Event
+        {
+            get
+            {
+                lock (_lock)
+                {
+                    return _event;
+                }
+            }
+            set
+            {
+                lock (_lock)
+                {
+                    _event = value;
+                }
+            }
```

Comment:
Why do we need the lock? `?.` shouldn't need it.

Reply:
The locks allows the behavior to be deterministic. I realized that if you change `Event` around the same time that `EndSuspend()` is running, then you might end up caching the action before it's changed, then invoke the old handler. Admittedly in our use case it's not happening, but that makes the class safe to reuse in another context, and shouldn't impact performance.

If I am not mistaken, the `?.` in `EndSuspend()` as well as the `(cb != null)` check in `Invoke()` are actually critical, and fixing an existing bug in some C# test. Because `Event` is a delegate it evaluates to `null` if there is no handler registered with it. And that happens if the user doesn't register any handler with `_renegotiationNeededEvent`, the event we use that `DelayedEvent` helper with.
https://codeblog.jonskeet.uk/2015/01/30/clean-event-handlers-invocation-with-c-6/

Reply:
Yeah, just saying that using `?.` doesn't require a lock. Determinism is good reason enough I suppose.